ragna/deploy/_engine.py

```diff
@@ -182,17 +182,25 @@ async def store_documents(
 
         streams = dict(ids_and_streams)
 
-        with self._database.get_session() as session:
-            documents = self._database.get_documents(
-                session, user=user, ids=streams.keys()
-            )
+        documents = self.get_documents(user, streams.keys())
 
         for document in documents:
             core_document = cast(
                 ragna.core.LocalDocument, self._to_core.document(document)
             )
             await core_document._write(streams[document.id])
 
+    def get_documents(
+        self, user: str, ids: Collection[uuid.UUID] | None = None
```

Comment:
In a function named `get_documents` it is not at all obvious what needs to be passed. Thus, just enforce keyword arguments so you don't end up with calls that are hard to read.

```suggestion
        self, *, user: str, ids: Collection[uuid.UUID] | None = None
```

Reply:
I [added keyword argument enforcement to both `get_document` and `get_documents`](https://github.com/Quansight/ragna/pull/547/commits/162a3ff53ef0f1616bd465ec43fbbe3e6e0bbc19#diff-8c3ee07a1ceeec05b8505832311b1e6ff5105b4badfe2040d388aace7d17a105). I assume this is what you wanted. Please let me know if it wasn't.